monitor/consul/watcher_test.go

```diff
@@ -41,24 +40,44 @@ func TestNew(t *testing.T) {
 }
 
 func TestWatcher_Watch(t *testing.T) {
-	w, err := New("xxx", "", "", 0, Item{})
-	require.NoError(t, err)
 	type args struct {
-		ctx context.Context
-		ch  chan<- []*change.Change
+		ctx    context.Context
+		cancel context.CancelFunc
+	}
+	withCancel := func() args {
+		ctx, cnl := context.WithCancel(context.Background())
+		return args{ctx: ctx, cancel: cnl}
 	}
 	tests := map[string]struct {
 		args    args
+		items   []Item
 		wantErr bool
 	}{
-		"missing context": {args: args{}, wantErr: true},
-		"missing chan":    {args: args{ctx: context.Background()}, wantErr: true},
+		"invalid item (missing type)": {
```

Comment:
New unit-test: 

in current master branch an Item with missing type would not cause an error, instead a segfault for nil pointer dereference happens.